lib/ship_fosdick/shipments_updater.rb

```diff
@@ -18,28 +19,23 @@ def initialize(files)
     # job number at index 5
     # run date at index 6
     def process
-      files.each do |file|
-        file.each_line do |line|
-          next if line.blank? || line.match(/(Ext Order #|TRAILER RECORD|SKU)/)
-          record = line.split(' ')
-          update_shipment(record)
-        end
+      rows = manifest_file.parse_content
+      rows.each do |record|
+        update_shipment(record)
       end
     end
 
     private
-
-    attr_reader :files
-
     def update_shipment(record)
       shipment = ::Spree::Shipment.find_by(number: record[0].strip)
```

Comment:
Also thought about that, however there is no other way really to fetch a shipment. We only have the shipment number from Fosdick. 

Ideally we inject this so we could do a "silent" update as well. i.e. without sending emails and triggering callbacks. Sometimes we need that :)